### PR TITLE
Meteors now leave behind a bit of ore

### PR DESCRIPTION
**Pull request by Plykiya**
## About the PR

Made meteors leave behind asteroid rocks after the explosion

## Why / Balance

idk, an admin told me to do it
flavor?

## Technical details

- Created WeightedSpawnEntityBehavior that takes a WeightedRandomEntityPrototype and spawns a number of the same entity between a given min and max, at a random offset from the final position of the entity.
- Spawning can be delayed if, for example, the entity has an explosive behavior that would destroy your newly spawned entity. If this happens, a temporary entity is created to act as a spawner.
- Created a spawn table of possible asteroid ores that the meteors will now leave behind with values chosen at whatever I randomly decided was a number
- Arbitrarily chosen spawn values that seemed appropriate for the size of the meteors

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**KABOOM VIDEO**
https://www.youtube.com/watch?v=tgqEwkIMV7o

## Breaking changes

**Changelog**

:cl:
- add: Meteors now leave behind asteroid rocks on impact.

### CODE REVIEW

**Comment by UbaserB** [comment]
This is actually very cool, woah..

**Comment by beck-thompson** [comment]
Maybe salvage will finally stay on station!

File: Content.Server/Destructible/Thresholds/Behaviors/WeightedSpawnEntityBehavior.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+using Robust.Shared.Spawners;
+
+namespace Content.Server.Destructible.Thresholds.Behaviors;
+
+[Serializable]
+[DataDefinition]
+public sealed partial class WeightedSpawnEntityBehavior : IThresholdBehavior
+{
+    [DataField(customTypeSerializer: typeof(PrototypeIdSerializer<WeightedRandomEntityPrototype>), required: true)]
+    public string WeightedSpawn = string.Empty;
```
Comment (metalgearsloth) [inline Content.Server/Destructible/Thresholds/Behaviors/WeightedSpawnEntityBehavior.cs:18]: Please use the new standards for this i.e. ProtoId

File: Content.Server/Destructible/Thresholds/Behaviors/WeightedSpawnEntityBehavior.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+using Robust.Server.GameObjects;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+using Robust.Shared.Spawners;
+
+namespace Content.Server.Destructible.Thresholds.Behaviors;
+
+[Serializable]
+[DataDefinition]
+public sealed partial class WeightedSpawnEntityBehavior : IThresholdBehavior
```
Comment (metalgearsloth) [inline Content.Server/Destructible/Thresholds/Behaviors/WeightedSpawnEntityBehavior.cs:15]: Needs documenting.

File: Content.Server/Destructible/Thresholds/Behaviors/WeightedSpawnEntityBehavior.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        var position = transform.GetMapCoordinates(uid);
+        Vector2 GetRandomVector() => new (system.Random.NextFloat(-SpawnOffset, SpawnOffset), system.Random.NextFloat(-SpawnOffset, SpawnOffset));
+        var entity = system.PrototypeManager.Index<WeightedRandomEntityPrototype>(WeightedSpawn).Pick(system.Random);
+        var amountToSpawn = system.Random.NextFloat(MinSpawn, MaxSpawn);
+
+        if (SpawnAfter != 0)
+        {
+            for (var i = 0; i < amountToSpawn; i++)
+            {
+                var spawner = system.EntityManager.SpawnEntity("TemporaryEntityForTimedDespawnSpawners", position.Offset(GetRandomVector()));
```
Comment (metalgearsloth) [inline Content.Server/Destructible/Thresholds/Behaviors/WeightedSpawnEntityBehavior.cs:56]: Please use ProtoId so this doesn't cause runtime exceptions and gets validated by CI.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Comment by Plykiya** [comment]
TryStopNukeOpsFromConstantlyFailing my NEMESIS

**Pull request merged**